auth/schema/mutations/register.py

```diff
@@ -25,23 +26,43 @@ def validate_password(password):
         password_valid = password_length.compare(8, len(password))
         if password_valid == False:
             raise ValidationError("Password must be at least 8 characters")
+        # add regex for making sure password has characters and symbols
         return password
 
     def validate_email(email):
-        email_validator = EmailValidator()
-        email_valid = email_validator.__call__(email)
+        email_validator = EmailValidator(email)
         if email_valid == False:
             raise ValidationError()
         return email
-        
+
+    '''Validate a username and make sure it contains only alphanumeric characters and special characters'''
+    def validate_username(username):
+        username_validator = RegexValidator(
+            r"(^[~0-9A-Za-z]*)",
+            "Username must contain only alphanumeric characters and special characters",
+            None,
+            re.IGNORECASE)
+        valid_username = username_validator(username)
+        if valid_username == True:
+            raise ValidationError()
+        return username
+    
+    def validate_name(name):
+        if name.isalnum() == False:
+            raise ValidationError("Name must contain only alphanumeric characters")
+        return name
+
```

Comment:
Great work. Now You have to separate the concerns of the method. For instance validations are not the sole responsibility of the mutations, you can create utils module that contains all the validation methods that you can use in the `RegisterUser` mutation,  what this does, is decreases coupling which enhances maintainability of the application, so in case you need to validate a username in another mutation, you can find it in say a utility module that contains validations. 
What do you think? @Basemera

Reply:
Sounds good.